Extract more functionality into the public api

danmichaelo opened the pull request:
I split it into two methods since I have a use case myself I want to have symmetric mappings both ways without touching the hierarchical mappings.

  
I also made the inference of skos:related from skos:relatedMatch optional, per #53

osma commented on skosify/infer.py:
```diff
@@ -24,7 +24,11 @@ def skos_topConcept(rdf):
-    """Infer skos:broader/skos:narrower (S25) but only keep skos:narrower on request."""
+    """Infer skos:broader/skos:narrower (S25) but only keep skos:narrower on request.
+
+    :param bool narrower: If set to False, skos:narrowMatch will not be added,
```
Shouldn't this be "skos:narrower will not be added"?

danmichaelo replied:
Good catch! Fixed

osma commented:
Other than the comment on the docstring, looks good to me. @nichtich ?

danmichaelo commented:
I found that I actually depended on `check_hierarchy` in one of my scripts, so I extracted functionality from that method into a new `check` namespace. Since the method was quite complex, I split it into three smaller methods at the same time. If you are not happy with the names of the methods, feel free to edit that!

I'm not sure if I like the re-calling of `setup_top_concepts(rdf, mark_top_concepts)` after `check.hierarchy_cycles()`, but I kept it for backwards compability. I guess it's there just in case the cycle breaking reveals new top concepts?

danmichaelo commented:
The latest commit is not really related to the rest of the PR, sorry about that.

osma commented:
> I'm not sure if I like the re-calling of setup_top_concepts(rdf, mark_top_concepts) after check.hierarchy_cycles(), but I kept it for backwards compability. I guess it's there just in case the cycle breaking reveals new top concepts?

That's exactly the reason. When I tested Skosify with a number of SKOS vocabularies back in 2011-2013 and wrote the EKAW and JoDS papers about it, there were some (can't remember which ones) that had a cycle on the top level. They were not detected as top concepts at first because all the concepts seemed to have a broader relationship.

Back then the Skosify processing was just a sequence of processing steps. Now that @nichtich and you have been pulling apart the processing steps and making them into a public API, the old assumptions are not valid anymore.

danmichaelo commented:
@nichtich , do you have a chance to look at this?

danmichaelo commented:
Rebased onto master
